pkg/controllers/binding/common.go

```diff
@@ -104,26 +110,18 @@ func ensureWork(c client.Client, workload *unstructured.Unstructured, overrideMa
 
 		workLabel := mergeLabel(clonedWorkload, workNamespace, binding, scope)
 
-		if clonedWorkload.GetKind() == util.DeploymentKind && (referenceRSP != nil || hasScheduledReplica) {
-			err = applyReplicaSchedulingPolicy(clonedWorkload, desireReplicaInfos[targetCluster.Name], util.ReplicasField)
+		if (hasScheduledReplica || referenceRSP != nil) && resourceInterpreter.HookEnabled(clonedWorkload, configv1alpha1.InterpreterOperationReviseReplica) {
```

Comment:
The `RSP` has been deprecated and should be cleaned up. I'll open an issue to track it.

Reply:
I have not used the `referenceRSP` check logic, this part of the code will be cleaned up separately after the PR merge, #1153